### ISSUE TITLE
Hidden FID data? hows qual browser doing it?

### ISSUE THREAD

**Issue by gjgfml** [compomics/ThermoRawFileParser#183]
Hi all, 

I'm posting not so much an issue  but more fishing for some insight (apologies if its not the appropriate forum). 
Below is a screenshot of same scan, m/z and rt of a data collected on an Orbitrap.
Looking at the same region in different software/processing tools, it quickly became aparent that the extra peaks are truncation wiggles (without negative peaks!! nve peaks get zeroed??)
Mzmine has some function/tool to "deal" with these ripples, but needs adjusting and parametrizing. 
My initial question is how is Qual browser doing this correction? 
It looks like its doing it on the fly (or maybe as it opens the files?) without any parametrization or user defined input. 
If so, it could imply that Qual browser is actually doing it in the time domain (FT, IFT, window functions etc ).
Is the time domain data actually recorded on the .raw files? Can we access it? We have a ton of tools we could leverage from the NMR comunity if this is the case. 

So tl;dr, do the .raw files contain hidden FIDs (I know it sounds almost like aconspiracy theory) and is it possible to extract it?

Hope you can help, 
Regards, 
G

![image](https://github.com/user-attachments/assets/01146eb1-6337-47d2-8217-d853aa17d05c)

**Comment by caetera** [comment]
Hi @gjgfml,
Please, ensure you have the same setting (enable/disable) for the **reference and exception** data while converting to mzML and viewing the file in QualBrowser. If you use ThermoRawFileParser or MSConvert, **reference and exception** peaks are included by default, you can exclude them (in TRFP) by providing `-x, --excludeExceptionData` key. QualBrowser, to the best of my knowledge, excludes these peaks by default. You can find a bit more details in this comment https://github.com/compomics/ThermoRawFileParser/issues/171#issuecomment-1733602484

As to FID data stored in the RAW file, I believe it is better to address this question to Thermo.
It is possible to switch to the "frequency" or "transient" signal in the Tune software (at least on tribrid instruments). However, I am not certain, if this is the "real" signal (i.e. directly from DAC) or the reconstructed one. Considering the amount of data in the FID and the speed of acquisition (up to 40 Hz in Orbitrap), it is likely that the FID signal never leaves the built-in computer.
The library TRFP uses for reading RAW files does not expose any API to access FID data, even if it is stored in the RAW file.

**Comment by caetera** [comment]
Out of my experience, almost every peak has an empty flag field, but these flags are set sometimes. If I recall, complete list of all possible flags can be found in QualBrowser help (cannot have access to it now).

A company named Spectoroswiss sells a device, that reads FID signal, but it plugs in parallel with original vendor acquisition system - https://spectroswiss.ch/hardware/. Maybe, it is something that might be relevant for you.
